### PR TITLE
DATAES-312 - Fix NullHandling.NULLS_LAST in query.sort

### PR DESCRIPTION
**Pull request by tedliang**
Fix [DATAES-312](https://jira.spring.io/browse/DATAES-312)

### CODE REVIEW

**Comment by tedliang** [comment]
It has been rebased to latest master. Please merge it at your earliest convenience.

**Comment by xhaggi** [comment]
@tedliang could you please rebase against master and squash your commits into a single one. Please adjust your commit message to include the issue id `DATAES-312 - Fix NullHandling.NULLS_LAST in query.sort`. See our contributing guide lines https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java [hunk 1]
```diff
@@ -420,6 +421,88 @@ public void shouldSortResultsGivenSortCriteria() {
 		assertThat(sampleEntities.getContent().get(0).getRate(), is(sampleEntity2.getRate()));
 	}
 
+	@Test
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java]: Please add the related issue ID as comment above `@Test`

File: src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java [hunk 2]
```diff
@@ -420,6 +421,88 @@ public void shouldSortResultsGivenSortCriteria() {
...
+				.build();
+		// when
+		Page<SampleEntity> sampleEntities = elasticsearchTemplate.queryForPage(searchQuery, SampleEntity.class);
+		// then
+		assertThat(sampleEntities.getTotalElements(), equalTo(3L));
+		assertThat(sampleEntities.getContent().get(0).getRate(), is(sampleEntity3.getRate()));
+		assertThat(sampleEntities.getContent().get(1).getMessage(), is(sampleEntity1.getMessage()));
+	}
+
+	@Test
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java]: same here

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java [hunk 3]
```diff
@@ -994,8 +997,15 @@ private SearchRequestBuilder prepareSearch(Query query) {
...
 		if (query.getSort() != null) {
 			for (Sort.Order order : query.getSort()) {
-				searchRequestBuilder.addSort(order.getProperty(),
-						order.getDirection() == Sort.Direction.DESC ? SortOrder.DESC : SortOrder.ASC);
+				FieldSortBuilder sort = SortBuilders.fieldSort(order.getProperty())
+						.order(order.getDirection().isDescending() ? SortOrder.DESC : SortOrder.ASC);
+				if (order.getNullHandling() == Sort.NullHandling.NULLS_FIRST) {
+					sort.missing("_first");
+				}
+				else if (order.getNullHandling() == Sort.NullHandling.NULLS_LAST) {
```
Comment (xhaggi) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java]: Please move `else if` on the same line with `}`, so `} else if(...) {`

**Comment by xhaggi** [comment]
@tedliang Thank you for contribution. I addressed the comments myself and merged the changes into master.

**Pull request closed without merging**